scenarios/test_codegen/test/rollback/Rollback_test.res

```diff
@@ -409,3 +411,736 @@ describe("Single Chain Simple Rollback", () => {
+        indexerMock.queryHistory(module(Entities.SimpleEntity)),
+      )),
+      (
+        [
+          {
+            Entities.SimpleEntity.id: "1",
+            value: "call-5",
+          },
+        ],
+        [
+          {
+            current: {
+              chain_id: 100,
+              block_timestamp: 101,
+              block_number: 101,
+              log_index: 2,
+            },
+            previous: undefined,
+            entityData: Set({
+              Entities.SimpleEntity.id: "1",
+              value: "call-0",
+            }),
+          },
+          {
+            current: {
+              chain_id: 1337,
+              block_timestamp: 101,
+              block_number: 101,
+              log_index: 1,
+            },
+            previous: Some({
+              chain_id: 100,
+              block_timestamp: 101,
+              block_number: 101,
+              log_index: 2,
+            }),
+            entityData: Set({
+              Entities.SimpleEntity.id: "1",
+              value: "call-1",
+            }),
+          },
+          {
+            current: {
+              chain_id: 1337,
+              block_timestamp: 101,
+              block_number: 101,
+              log_index: 2,
+            },
+            previous: Some({
+              chain_id: 1337,
+              block_timestamp: 101,
+              block_number: 101,
+              log_index: 1,
+            }),
+            entityData: Set({
+              Entities.SimpleEntity.id: "1",
+              value: "call-2",
+            }),
+          },
+          {
+            current: {
+              chain_id: 1337,
+              block_timestamp: 102,
+              block_number: 102,
+              log_index: 2,
+            },
+            previous: Some({
+              chain_id: 1337,
+              block_timestamp: 101,
+              block_number: 101,
+              log_index: 2,
+            }),
+            entityData: Set({
+              Entities.SimpleEntity.id: "1",
+              value: "call-3",
+            }),
+          },
+          {
+            current: {
+              chain_id: 100,
+              block_timestamp: 102,
+              block_number: 102,
+              log_index: 2,
+            },
+            previous: Some({
+              chain_id: 1337,
+              block_timestamp: 102,
+              block_number: 102,
+              log_index: 2,
+            }),
+            entityData: Set({
+              Entities.SimpleEntity.id: "1",
+              value: "call-4",
+            }),
+          },
+          {
+            current: {
+              chain_id: 1337,
+              block_timestamp: 102,
+              block_number: 102,
+              log_index: 4,
+            },
+            // FIXME: This looks wrong
+            previous: Some({
+              chain_id: 1337,
+              block_timestamp: 102,
+              block_number: 102,
+              log_index: 2,
+            }),
```

Comment:
Insert entity history function should search for previous by serial not by timestamp